### PR TITLE
[DOC] Add stale issue policy to CONTRIBUTING

### LINKED ISSUE #258: Add Stale tag to Issues
### Summary
I'd like to propose adding a stale tag to an issue that meets certain criteria, with the intent of closing them if they remain stale over some time period.

### Additional Detail
I don't feel strongly about our criteria for a stale issue or how long it remains stale before closure, but I think it would be help keep the issue tracker from getting bloated. As an examples for issues that I think are stale:
1. #144 has little discussion since October of last year, and has never had a PR directly address. Discussion seemed to converge on adding documentation.
2. #50 has not been referenced since October as well, and is probably best broken into smaller issues like @tsalo has done in #249, #250 (Thanks @tsalo!)
The other advantage it has is that if we see that a conversation has stalled, we can think about why and decide if there is a way to either reignite discussion or resolve it. For example, it would be useful in my opinion to close #144 and replace it with some issues on documentation, as well as notes in the reliability analysis. Similarly, I think #50 could easily be replaced with @tsalo's new issues, and additional ones as we refactor more deeply. 

### Next Steps
- [x] I'd like to hear back from the maintainers about their thoughts on this, especially if there's already a policy in place that I'm just not aware of
- [x] I'd like to hear from contributors on whether they think this would make it easier to read through the issue tracker
- [x] If we decide to implement a stale tag, we should define a clear policy for what a stale issue would be
- [x] If we decide to implement a stale tag, we should define a policy for closing a stale issue if it remains stale for a certain period, including what closing comments should state about its closure (e.g., should they point to new issues that are spawned from that thread?)
- [x] If we decide to implement the tag, we should indicate policy on stale issues in CONTRIBUTING (Would be completed by #279)
- [ ] Add stale-bot to the repository

### PR DESCRIPTION
**Pull request by jbteves**
Partially closes #258

Changes proposed in this pull request:

- Adds policy for stale issues to CONTRIBUTING

### CODE REVIEW

**Comment by jbteves** [comment]
I'm not wild about where I put this policy in, but I couldn't think of a better place to describe it other than adding that label to the issue labels section. I'd love to get some input on this.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -65,6 +65,11 @@ is difficult to describe as one unit of work, please consider splitting it into
     Issues are assigned [labels](#issue-labels) which explain how they relate to the overall project's
     goals and immediate next steps.
 
+    Sometimes issues may not produce action items, and conversation will stall after a few months. 
+    When this happens, they may be marked stale by [stale-bot][link_stale-bot], 
+    and will be closed after a week unless there is more discussion.
```
Comment (emdupre) [inline CONTRIBUTING.md]: Can we add a note here about how folks can jump in and keep the issues from closing if they think the discussion isn't resolved ?

Comment (jbteves) [reply CONTRIBUTING.md]: Is cfe50d8 what you would like?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -65,6 +65,12 @@ is difficult to describe as one unit of work, please consider splitting it into
     Issues are assigned [labels](#issue-labels) which explain how they relate to the overall project's
     goals and immediate next steps.
 
+    Sometimes issues may not produce action items, and conversation will stall after a few months. 
+    When this happens, they may be marked stale by [stale-bot][link_stale-bot], 
+    and will be closed after a week unless there is more discussion. 
+    This helps us keep the issue tracker organized.
+    Please feel free to discuss if you feel an issue should remain open!
```
Comment (emdupre) [inline CONTRIBUTING.md]: Super nit-picky, feel free to ignore

```suggestion
    Any new discussion on the issue will remove the `stable` label, and prevent it from closing.
    So, if theres's a discussion you think it not yet resolved, please jump in !
```

**Comment by jbteves** [comment]
Great, thank you! Since this is just one commit, is there any opposition to my merging without squashing? Or would you like the PR name as the short commit message?

**Comment by emdupre** [comment]
No I think it's fine to just merge in as-is. Eventually we'll have to get principled about this, but I'm not worried just yet :smile_cat:

**Pull request merged**